src/main/java/com/waverleysoftware/pageobjects/GoogleAllPage.java

```diff
@@ -0,0 +1,31 @@
+package com.waverleysoftware.pageobjects;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+import static com.waverleysoftware.listemers.TestListener.getSoftAssert;
+
+/**
+ * A simple class which describes the behavior of google "all" page.
+ */
+public class GoogleAllPage extends BasePage {
+
+    private final List<WebElement> listOfLinks = findAll(By.cssSelector("#rso h3 > a"));
+
+    public GoogleAllPage(final WebDriver driver) {
+        super(driver);
+    }
+
+    public GoogleAllPage listSize(final int size) {
+        getSoftAssert().assertThat(listOfLinks).as("links size").hasSize(size);
```

Comment:
page objects are not intended for verifications... assertions should be made on tests level...